osf/migrations/0135_migrate_registrations_to_osf_registries_provider.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.13 on 2018-07-18 13:51
+from __future__ import unicode_literals
+import logging
+from itertools import chain
+
+from django.db import migrations
+from django.db.models import Q
+from django.core.paginator import Paginator
+from django_bulk_update.helper import bulk_update
+
+logger = logging.getLogger(__file__)
+
+def migrate_existing_registrations_into_osf_registries(state, schema):
+    AbstractProvider = state.get_model('osf', 'abstractprovider')
+    AbstractNode = state.get_model('osf', 'abstractnode')
+    DraftRegistration = state.get_model('osf', 'draftregistration')
+
+    try:
+        osf_registries = AbstractProvider.objects.get(_id='osf', type='osf.registrationprovider')
+    except AbstractProvider.DoesNotExist:
+        # Allow test / local dev DBs to pass
+        logger.warn('Unable to find OSF Registries provider - assuming test environment.')
+        pass
+    else:
+        public_registrations_list = AbstractNode.objects.filter(
+            type='osf.registration',
+            is_public=True,
+            is_deleted=False,
+        ).filter(
+            Q(Q(embargo__isnull=True) | ~Q(embargo__state='unapproved')) &
+            Q(Q(retraction__isnull=True) | ~Q(retraction__state='approved'))
+        ).order_by('id')
+
+        draft_registrations_list = DraftRegistration.objects.filter(deleted__isnull=True).order_by('id')
+        regs_to_migrate = chain(public_registrations_list, draft_registrations_list)
+        paginated_registrations = Paginator(list(regs_to_migrate), 1000)
+        total_registrations = public_registrations_list.count() + draft_registrations_list.count()
+        total_migrated = 0
+        logger.info('Migrating {} public and draft registrations into OSF Registries...'.format(total_registrations))
+        for page in paginated_registrations.page_range:
+            targets = []
+            for reg in paginated_registrations.page(page).object_list:
+                reg.provider = osf_registries
+                total_migrated += 1
+            bulk_update(targets, update_fields=['provider'])
```

Comment:
Additionally, I'd be surprised if [the `bulk_update` code](https://github.com/aykut/django-bulk-update/blob/master/django_bulk_update/helper.py#L116) supports cross-table updates, since the schema is inferred from the first object it sniffs.